### PR TITLE
feat: Add support for encoding parameter on read_excel

### LINKED ISSUE #23444: 'ascii' codec can't decode byte 0xd1 in position 2: ordinal not in range(128)
#### Code Sample, a copy-pastable example if possible

```python
dataframe = pd.read_excel(
                StringIO(self.file_stream), sheet, na_values=['undefined', 'NaN'], header=None, keep_default_na=False
            )
```
#### Problem description

I'm trying to read a xls file with `read_excel()` method and it throws the error on the title. If I try to read the file with xlrd lib I can fix the error by providing the parameter `encoding_override` with the file encoding. I've seen some Stackoverflow answers and all of them recommend using an `encoding` parameter, which doesn't exist. Why don't the implement an encoding parameter for the `read_excel()` method, and just use it as `encoding_override` when reading the file with xlrd?

**Note**: We receive a lot of issues on our GitHub tracker, so it is very possible that your issue has been posted before. Please check first before submitting so that we do not have to handle and close duplicates!

**Note**: Many problems can be resolved by simply upgrading `pandas` to the latest version. Before submitting, please check if that solution works for you. If possible, you may want to check if `master` addresses this issue, but that is not necessary.

For documentation-related issues, you can check the latest versions of the docs on `master` here:

https://pandas-docs.github.io/pandas-docs-travis/

If the issue has not been resolved there, go ahead and file it in the issue tracker.

#### Output of ``pd.show_versions()``

<details>

[paste the output of ``pd.show_versions()`` here below this line]

</details>

### PR DESCRIPTION
**Pull request by mau21mau**
- [ ] closes #23444
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/io/excel.py [hunk 1]
```diff
@@ -400,9 +400,15 @@ def __init__(self, io, **kwds):
                     pass
 
             data = io.read()
-            self.book = xlrd.open_workbook(file_contents=data)
```
Comment (jreback) [inline pandas/io/excel.py]: you can just wrap this into a kwargs, adding only if its not None, then pass ``**kwargs`` on construction.

**Review by jreback — CHANGES_REQUESTED** [review]
this would absolutely need a test

File: pandas/io/excel.py (hunk 1 above)
Comment (mau21mau) [reply pandas/io/excel.py]: I agree. I did a new commit on the matter.

**Comment by jreback** [comment]
this needs tests, also pls merge master

**Comment by jreback** [comment]
needs tests. ping if you want to keep working.

**Pull request closed without merging**